src/Chapter2.hs

```diff
@@ -841,7 +872,8 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate offset list = subList offset (length list + (offset - 1)) $ cycle list
```

Comment:
Interesting solution with `subList`! 💯 

Unfortunately, `cycle` fails in runtime on empty lists :recycle: So you need to handle this case separately for this function to work properly.

Also, even hat's a correct solution, though, it does a little bit more than work when the given number is big and the list is small. You will cycle through the list a lot. Maybe you can think of a way to optimize this :slightly_smiling_face:

Reply:
Thanks for the suggestion, I took another shot at it